Tests/AdyenTests/Adyen Tests/Components/Affirm/AffirmComponentTests.swift

```diff
@@ -0,0 +1,210 @@
+            XCTAssertEqual(details.shopperName?.firstName, "Katrina")
+            XCTAssertEqual(details.shopperName?.lastName, "Del Mar")
+            XCTAssertEqual(details.telephoneNumber, "2025550146")
+            XCTAssertEqual(details.emailAddress, "katrina@mail.com")
+            XCTAssertEqual(details.billingAddress, expectedBillingAddress)
+            XCTAssertEqual(details.deliveryAddress, expectedDeliveryAddress)
+            
+            sut.stopLoadingIfNeeded()
+            didSubmitExpectation.fulfill()
+        }
+                
+        wait(for: .seconds(1))
+        
+        let view: UIView = sut.viewController.view
+        
+        let firstNameView: FormTextInputItemView = try XCTUnwrap(view.findView(by: "AdyenComponents.AffirmComponent.firstNameItem"))
+        populate(textItemView: firstNameView, with: "Katrina")
+        
+        let lastNameView: FormTextInputItemView = try XCTUnwrap(view.findView(by: "AdyenComponents.AffirmComponent.lastNameItem"))
+        populate(textItemView: lastNameView, with: "Del Mar")
+        
+        let phoneNumberView: FormPhoneNumberItemView = try XCTUnwrap(view.findView(by: "AdyenComponents.AffirmComponent.phoneNumberItem"))
+        populate(textItemView: phoneNumberView, with: "2025550146")
+        
+        let emailView: FormTextInputItemView = try XCTUnwrap(view.findView(by: "AdyenComponents.AffirmComponent.emailItem"))
+        populate(textItemView: emailView, with: "katrina@mail.com")
+        
+        // ===================== BILLING ADDRESS =====================
+        
+        let billingAddressView: FormVerticalStackItemView<FormAddressItem> = try XCTUnwrap(view.findView(by: "AdyenComponents.AffirmComponent.billingAddressItem"))
+        fill(addressView: billingAddressView, with: expectedBillingAddress)
+        
+        // ===================== TOGGLE ITEM ==========================
+        
+        let deliveryAddressToggleView: FormToggleItemView! = try XCTUnwrap(view.findView(by: "AdyenComponents.AffirmComponent.deliveryAddressToggleItem"))
+        deliveryAddressToggleView.switchControl.isOn = true
+        deliveryAddressToggleView.switchControl.sendActions(for: .valueChanged)
+        
+        // ===================== DELIVERY ADDRESS =====================
+        
+        let deliveryAddressView: FormVerticalStackItemView<FormAddressItem> = try XCTUnwrap(view.findView(by: "AdyenComponents.AffirmComponent.deliveryAddressItem"))
+        fill(addressView: deliveryAddressView, with: expectedDeliveryAddress)
+        
+        let submitButton: UIControl = try XCTUnwrap(view.findView(by: "AdyenComponents.AffirmComponent.payButtonItem.button"))
+        submitButton.sendActions(for: .touchUpInside)
+        
+        waitForExpectations(timeout: 10, handler: nil)
+    }
+    
+    // MARK: - Private
+    
+    private func fill(addressView: FormVerticalStackItemView<FormAddressItem>, with address: PostalAddress) {
```

Comment:
this could be public :)

Reply:
Yeah, gonna move this to an `XCTestCase` extension so it's available across tests